### PR TITLE
Links file module, marks fopen and fclose deprecation in compiler.go

### PR DESCRIPTION
**Pull request by supreetsingh10**
and builtin_fn_io.go

### CODE REVIEW

File: pkg/eval/mods/file/file.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+// File should be closed after opening using close
+// Example
+//  ```elvish-transcript
+// ~> cat a.txt
+// Some text in file
+// f = (file:open a.txt)
+// ~> cat < $f
+// Some text in file
+// ~> file:close $f
+// ```
```
Comment (xiaq) [inline pkg/eval/mods/file/file.go:24]: This doc (and the one for `close` below) need more work; you could also have just copied the existing doc from `fopen` and `fclose`. I'll follow up with a commit after merging this.

File: pkg/eval/mods/file/file.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+func close(f vals.File) error {
+	return f.Close()
+}
+
+var Ns = eval.NsBuilder{}.AddGoFns("file:", fns).Ns()
+
+var fns = map[string]interface{}{
+	"close": close,
+	"open":  open,
+}
```
Comment (xiaq) [inline pkg/eval/mods/file/file.go:47]: Follow the convention in other module implementations, and put these at the beginning of the file.

File: pkg/eval/mods/file/file_test.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package file
+
+import (
+	"testing"
+
+	"src.elv.sh/pkg/eval"
+	"src.elv.sh/pkg/eval/errs"
+	"src.elv.sh/pkg/eval/evaltest"
```
Comment (xiaq) [inline pkg/eval/mods/file/file_test.go:8]: Follow the convention in other module implementations and use a dot import (see https://golang.org/ref/spec#Import_declarations) to import `evaltest`.

**Review by xiaq — APPROVED** [review]
Some minor feedback on the code style. I'll merge this PR and tweak the code a little bit in a followup commit.

File: pkg/eval/mods/file/file.go (hunk 1 above)
Comment (supreetsingh10) [reply pkg/eval/mods/file/file.go:24]: I understand, thank you for suggesting this.

File: pkg/eval/mods/file/file.go (hunk 2 above)
Comment (supreetsingh10) [reply pkg/eval/mods/file/file.go:47]: Alright, I used str.go as a reference, anyway I will take care of this in the future.

**Pull request merged**